docs/literate/src/files/structured_mesh_mapping.jl

```diff
@@ -0,0 +1,202 @@
+analysis_callback = AnalysisCallback(semi, interval=analysis_interval)
+
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+stepsize_callback = StepsizeCallback(cfl=0.9)
+
+callbacks = CallbackSet(analysis_callback,
+                        alive_callback,
+                        stepsize_callback)
+
+# Running the simulation
+sol = solve(ode, CarpenterKennedy2N54(williamson_condition=false),
+            dt=1.0, # solve needs some value here but it will be overwritten by the stepsize_callback
+            save_everystep=false, callback=callbacks);
+
+using Plots
+plot(sol)
+#-
+pd = PlotData2D(sol)
+plot(pd["p"])
+plot!(getmesh(pd))
+
+
+# ## Mesh directly defined by the transformation mapping
+# As mentioned before, you can also define the domain for a `StructuredMesh` by directly setting up
+# a transformation mapping. Here, we want to present a nice mapping, which is often used to test
+# free-stream preservation. Exact free-stream preservation is a crucial property of any numerical
+# method on curvilinear grids. The mapping is a reduced 2D version of the mapping described in
+# [Rueda-Ramírez et al. (2021), p.18](https://arxiv.org/abs/2012.12040).
+
+using OrdinaryDiffEq
+using Trixi
+
+equations = CompressibleEulerEquations2D(1.4)
+
+# As mentioned, this mapping is used for testing free-stream preservation. So, we use a constant
+# initial condition.
+initial_condition = initial_condition_constant
+
+solver = DGSEM(polydeg=3, surface_flux=flux_lax_friedrichs)
+
+# We define the transformation mapping with variables in $[-1, 1]$ as described in
+# Rueda-Ramírez et al. (2021), p.18 (reduced to 2D):
+function mapping(xi_, eta_)
+  ## Transform input variables between -1 and 1 onto [0,3]
+  xi = 1.5 * xi_ + 1.5
+  eta = 1.5 * eta_ + 1.5
+
+  y = eta + 3/8 * (cos(1.5 * pi * (2 * xi - 3)/3) *
+                   cos(0.5 * pi * (2 * eta - 3)/3))
+
+  x = xi + 3/8 * (cos(0.5 * pi * (2 * xi - 3)/3) *
+                  cos(2 * pi * (2 * y - 3)/3))
+
+  return SVector(x, y)
+end
+
+cells_per_dimension = (16, 16)
+
+# Instead of a tuple of boundary functions, the `mesh` now has the mapping as its parameter.
+mesh = StructuredMesh(cells_per_dimension, mapping)
+
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver)
+
+tspan = (0.0, 1.0)
+ode = semidiscretize(semi, tspan)
+
+analysis_interval = 100
+analysis_callback = AnalysisCallback(semi, interval=analysis_interval)
+
+alive_callback = AliveCallback(analysis_interval=analysis_interval)
+
+stepsize_callback = StepsizeCallback(cfl=0.9)
+
+callbacks = CallbackSet(analysis_callback, alive_callback,
+                        stepsize_callback)
+
+sol = solve(ode, CarpenterKennedy2N54(williamson_condition=false),
+            dt=1.0, # solve needs some value here but it will be overwritten by the stepsize_callback
+            save_everystep=false, callback=callbacks);
+
+using Plots
+pd = PlotData2D(sol)
+plot(pd["rho"])
+plot!(getmesh(pd))
```

Comment:
@ranocha Is it possible to change the plot such as we don't just see the density but the difference of the approximated density to `1` in a log_10 scale?

Reply:
Thanks. That was a bit tricky for `StructuredMesh` since there `pd.x` and `pd.y` are matrices as well. That is not supported for `heatmap`.
Now, I used Trixi's plot function with the parameter `solution_variables` and defined my own variable there.
```
@inline cons2denserror(u, ::CompressibleEulerEquations2D) = SVector(abs(u[1] - 1.0), u[2:4]...)
Trixi.varnames(::typeof(cons2denserror), ::CompressibleEulerEquations2D) = ("error_density", "rho_v1", "rho_v2", "rho_e")
pd = PlotData2D(sol; solution_variables=cons2denserror)

using Plots
plot(pd["error_density"])
plot!(getmesh(pd))
```
I'm a little confused that that doesn't work without `u[2:4]...` but I think that's acceptable.

I left out the `log10` because some values are exactly 0 and `plot` only accepts finite values. But I think it is understandable the way as it is now.
![plot](https://user-images.githubusercontent.com/74359358/158831520-396a37f3-b2db-4c37-8789-141f4de979d9.png)